src/optionals/http/http.c

```diff
@@ -609,6 +613,428 @@ static Value post(DictuVM *vm, int argCount, Value *args) {
+                    continue;
+                }
+
+                if (!IS_STRING(entry->value)) {
+                    runtimeError(vm, "HTTP client option \"keyFile\" value must be a string");
+                    return abstract;
+                }
+
+                char *keyFile = AS_STRING(entry->value)->chars;
+                if (keyFile[0] == '\0') {
+                    continue;
+                }
+                
+                curl_easy_setopt(httpClient->curl, CURLOPT_SSLKEY, keyFile);
+            } else if (strstr(key, "certFile")) {
+                if (IS_EMPTY(entry->value)) {
+                    continue;
+                }
+
+                if (!IS_STRING(entry->value)) {
+                    runtimeError(vm, "HTTP client option \"certFile\" value must be a string");
+                    return abstract;
+                }
+
+                char *certFile = AS_STRING(entry->value)->chars;
+                if (certFile[0] == '\0') {
+                    continue;
+                }
+
+                curl_easy_setopt(httpClient->curl, CURLOPT_SSLCERT, certFile);
+            } else if (strstr(key, "keyPasswd")) {
+                if (IS_EMPTY(entry->value)) {
+                    continue;
+                }
+
+                if (!IS_STRING(entry->value)) {
+                    runtimeError(vm, "HTTP client option key \"keyPasswd\" value must be a string");
+                    return abstract;
+                }
+
+                char *keyPasswd = AS_STRING(entry->value)->chars;
+                if (keyPasswd[0] == '\0') {
+                    continue;
+                }
+
+                curl_easy_setopt(httpClient->curl, CURLOPT_KEYPASSWD, keyPasswd);
+            }
+        }
+    }
+
+    abstract->data = httpClient;
+
+    /**
+     * Setup HTTP object methods
+     */
+    defineNative(vm, &abstract->values, "get", httpClientGet);
+    defineNative(vm, &abstract->values, "post", httpClientPost);
+    defineNative(vm, &abstract->values, "setTimeout", httpClientSetTimeout);
+    defineNative(vm, &abstract->values, "setHeaders", httpClientSetHeaders);
+    defineNative(vm, &abstract->values, "setInsecure", httpClientSetInsecure);
+    defineNative(vm, &abstract->values, "setKeyFile", httpClientSetKeyFile);
+    defineNative(vm, &abstract->values, "setCertFile", httpClientSetCertFile);
+    defineNative(vm, &abstract->values, "setKeyPass", httpClientSetKeyPass);
+    pop(vm);
+
+    return abstract;
+}
+
+static Value newClient(DictuVM *vm, int argCount, Value *args) {
+    if (argCount > 1) {
+        runtimeError(vm, "newClient() takes 1 argument (%d given)", argCount);
+        return EMPTY_VAL;
+    }
+
+    if (!IS_DICT(args[0])) {
+        runtimeError(vm, "Options dict passed to newClient() must be a dict.");
+        return EMPTY_VAL;
+    }
+
+    ObjDict *opts = AS_DICT(args[0]);
+
+    ObjAbstract *hc = newHttpClient(vm, opts);
+    return newResultSuccess(vm, OBJ_VAL(hc));
```

Comment:
I wonder, do we actually need to return a `Result<>` type here if we only ever return a successful result?

Reply:
Updated to not return the Result type. Tests, examples, and docs updated. Good call. That makes a lot more sense. Thanks!